capsules/src/console.rs

```diff
@@ -204,15 +205,18 @@ impl<'a, U: UART> Client for Console<'a, U> {
         // Either print more from the AppSlice or send a callback to the
         // application.
         self.tx_buffer.replace(buffer);
-        self.in_progress.take().map(|appid| {
+        self.in_progress.get().map(|appid| {
+            self.in_progress.set(None);
```

Comment:
On second look, it seems like needing to .set(None) is not the normal case.

I'd still very much prefer a common paradigm where we can just call .map() on anything rather than having some .get().map() and some .map()